main/Main.hs

```diff
@@ -362,21 +293,40 @@ replyTo _ HistoryRequest{} replyHeader state = do
         }
   return (state, reply)
 
-handleComm :: (Message -> IO ()) -> KernelState -> Message -> MessageHeader -> IO KernelState
-handleComm replier kernelState req replyHeader = do
+handleComm :: (Message -> IO ()) -> KernelState -> Message -> MessageHeader -> Interpreter KernelState
```

Comment:
This is pretty dense code, even if it is pretty clear. Could you describe what `handleComm` does (in a comment) and maybe add comments in the function itself if there is anything you think is tricky and needs explaining?